jetty-ee10/jetty-ee10-proxy/src/main/java/org/eclipse/jetty/ee10/proxy/AsyncMiddleManServlet.java

```diff
@@ -610,7 +612,10 @@ public boolean offer(ByteBuffer content, Callback callback)
         {
             if (_log.isDebugEnabled())
                 _log.debug("{} proxying content to downstream: {} bytes {}", getRequestId(clientRequest), content.remaining(), callback);
-            return chunks.offer(new Chunk(content, callback));
+            try (AutoLock ignored = lock.lock())
```

Comment:
What is the benefit of this solution, rather than just using something like `ConcurrentLinkedDeque` for the `chunks` ?